ui-tests/tests/file-selection.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { test } from '@jupyterlab/galata';
+import { expect } from '@playwright/test';
+import path from 'path';
+import { extractFile } from './utils';
+
+const baseRepositoryPath = 'test-repository-dirty.tar.gz';
+test.use({ autoGoto: false });
+
+test.describe('File selection for normal staging', () => {
+  test.beforeEach(async ({ baseURL, page, tmpPath }) => {
+    await extractFile(
+      baseURL,
+      path.resolve(__dirname, 'data', baseRepositoryPath),
+      path.join(tmpPath, 'repository.tar.gz')
+    );
+
+    // URL for merge conflict example repository
+    await page.goto(`tree/${tmpPath}/test-repository`);
+  });
+
+  test('should select two files with ctlr-click', async ({ page }) => {
+    await page.sidebar.openTab('jp-git-sessions');
+    await page.click('button:has-text("Changes")');
+
+    // Click another_file.txt
+    await page.locator('#jp-git-sessions >> text=another_file.txt').click();
+    // Control-click master_file.ts
+    await page.locator('#jp-git-sessions >> text=master_file.ts').click({
+      modifiers: ['Control', 'Meta']
+    });
+
+    const selectedFileCount = await page.locator('[data-test-selected=true]').count();
+    expect(selectedFileCount).toBeGreaterThanOrEqual(2);
```

Comment:
Let's be strict here as the number of files should be deterministic (in the opposite to the Shift case).

```suggestion
    expect(selectedFileCount).toEqual(2);
```